cli/templateversionpresets.go

```diff
@@ -0,0 +1,137 @@
+package cli
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/cli/cliui"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/serpent"
+)
+
+func (r *RootCmd) templateVersionPresets() *serpent.Command {
+	cmd := &serpent.Command{
+		Use:     "presets",
+		Short:   "Manage presets of the specified template version",
+		Aliases: []string{"preset"},
+		Long: FormatExamples(
+			Example{
+				Description: "List presets of a specific template version",
+				Command:     "coder templates versions presets list my-template my-template-version",
+			},
+		),
+		Handler: func(inv *serpent.Invocation) error {
+			return inv.Command.HelpHandler(inv)
+		},
+		Children: []*serpent.Command{
+			r.templateVersionPresetsList(),
+		},
+	}
+
+	return cmd
+}
+
+func (r *RootCmd) templateVersionPresetsList() *serpent.Command {
+	defaultColumns := []string{
+		"name",
+		"parameters",
+		"default",
+		"prebuilds",
+	}
+	formatter := cliui.NewOutputFormatter(
+		cliui.TableFormat([]templateVersionPresetRow{}, defaultColumns),
+		cliui.JSONFormat(),
+	)
+	client := new(codersdk.Client)
+	orgContext := NewOrganizationContext()
+
+	cmd := &serpent.Command{
+		Use: "list <template> <version>",
+		Middleware: serpent.Chain(
+			serpent.RequireNArgs(2),
+			r.InitClient(client),
+		),
+		Short:   "List all the presets of the specified template version",
+		Options: serpent.OptionSet{},
+		Handler: func(inv *serpent.Invocation) error {
+			organization, err := orgContext.Selected(inv, client)
+			if err != nil {
+				return xerrors.Errorf("get current organization: %w", err)
+			}
+
+			template, err := client.TemplateByName(inv.Context(), organization.ID, inv.Args[0])
+			if err != nil {
+				return xerrors.Errorf("get template by name: %w", err)
+			}
+
+			version, err := client.TemplateVersionByName(inv.Context(), template.ID, inv.Args[1])
+			if err != nil {
+				return xerrors.Errorf("get template version by name: %w", err)
+			}
+
+			presets, err := client.TemplateVersionPresets(inv.Context(), version.ID)
+			if err != nil {
+				return xerrors.Errorf("get template versions presets by template version: %w", err)
+			}
+
+			if len(presets) == 0 {
+				return xerrors.Errorf("no presets found for template %q and template-version %q", template.Name, version.Name)
```

Comment:
Should this be an error? Or just an empty list?

Reply:
True, doesn't need to be an error. Addressed in: https://github.com/coder/coder/pull/18910/commits/641dc96e63dccee8c2984b3d946231269cc50584